### PR TITLE
[dev] AB#51574: Button Upgrade

### PR DESCRIPTION
**Pull request by cbrewer1689**

### CODE REVIEW

File: docs/src/global/exampleFrame.jsx
Comment (groberts314) [inline docs/src/global/exampleFrame.jsx:14]: Sigh.  Too bad ESLINT isn't smart about props used in the styles function.  But it's probably difficult to detect that.

File: docs/src/inputs/button/button.jsx
Comment (groberts314) [inline docs/src/inputs/button/button.jsx:67]: Types...?  😕 
Types of what?
Is this perhaps a copy-and-paste error, and this is meant to say something else?

File: docs/src/inputs/button/button.jsx
Comment (groberts314) [inline docs/src/inputs/button/button.jsx:89]: Same here.  I'll refrain from commenting on everyone, but please double-check all of these on this page.  Were you perhaps intending to keep some of the original "Subheader" text, and copied a `<Typography>` element from somewhere but then forgot to update the inner text?

File: docs/src/inputs/button/devSandbox/examples/exampleOutlineColors.jsx
Comment (groberts314) [inline docs/src/inputs/button/devSandbox/examples/exampleOutlineColors.jsx:22]: **_TOTAL NAB_** and 100% discretionary to do anything here.  Should we show `title` props on the `<Icon>`s in these examples?

File: src/inputs/button/__test__/button.test.js
Comment (groberts314) [inline src/inputs/button/__test__/button.test.js:91]: Ooh the grammar hurts here! ;-)
How about "Should add expected class when \'disable\' prop is true" ?

File: src/inputs/button/buttonConstants.js
Comment (groberts314) [inline src/inputs/button/buttonConstants.js:63]: NAB.  Should we add a documentation comment for `target` prop?

**Review by groberts314 — CHANGES_REQUESTED** [review]
Overall looks pretty good.  A few things here and there, mostly in the docs.  Updating HC for all the `disabled` => `disable` might be a bit of a chore! ;-)

**Comment by groberts314** [comment]
We've got unit test failures:

https://dev.azure.com/saddlebackchurch/Church%20Management/_build/results?buildId=37853&view=logs&j=fd490c07-0b22-5182-fac9-6d67fe1e939b&t=ec339c2b-c439-539d-f401-d5569bbc95f9

```
Summary of all failing tests
FAIL src/dataDisplay/personPanel/__test__/personPanelDetailsActionButton.test.js
  ● <personPanelDetailsActionButton /> › Should render with the button classes

    expect(received).toEqual(expected) // deep equality

    Expected: true
    Received: false

      58 |         root = wrapper.find('button');
      59 | 
    > 60 |         expect(root.hasClass(/(Button)-(outlined)-(\d+)/)).toEqual(true);
         |                                                            ^
      61 |     });
      62 | 
      63 |     it('Should render with the prompt classes', () => {

      at Object.<anonymous> (src/dataDisplay/personPanel/__test__/personPanelDetailsActionButton.test.js:60:60)

FAIL src/inputs/button/__test__/button.test.js
  ● <Button /> › Should render with proper classNames

    expect(received).toEqual(expected) // deep equality

    Expected: true
    Received: false

      72 | 
      73 |         expect(rootNode.hasClass('cmui')).toEqual(true);
    > 74 |         expect(rootNode.hasClass('button')).toEqual(true);
         |                                             ^
      75 |         expect(rootNode.hasClass(/(Button)-(root)-(\d+)/)).toEqual(true);
      76 |         expect(rootNode.hasClass(props.className)).toEqual(true);
      77 |     });

      at Object.<anonymous> (src/inputs/button/__test__/button.test.js:74:45)

Test Suites: 2 failed, 23 passed, 25 total
Tests:       2 failed, 1 todo, 251 passed, 254 total
Snapshots:   0 total
Time:        45.904 s
Ran all test suites.
```

File: docs/src/global/exampleFrame.jsx
Comment (cbrewer1689) [reply docs/src/global/exampleFrame.jsx:14]: Yeah, maybe in the future someone will fix this.

File: docs/src/inputs/button/button.jsx
Comment (cbrewer1689) [reply docs/src/inputs/button/button.jsx:67]: image

File: docs/src/inputs/button/button.jsx
Comment (cbrewer1689) [reply docs/src/inputs/button/button.jsx:89]: I posted a picture above that shows what is below the text "Types."

Comment (cbrewer1689) [reply docs/src/inputs/button/button.jsx:89]: It's following Design's UI docs.

File: docs/src/inputs/button/button.jsx
Comment (groberts314) [reply docs/src/inputs/button/button.jsx:67]: Makes sense.  Thanks!  Resolving this.

File: docs/src/inputs/button/button.jsx
Comment (groberts314) [reply docs/src/inputs/button/button.jsx:89]: Yep, got it.  Resolving.

File: src/inputs/button/__test__/button.test.js
Comment (groberts314) [reply src/inputs/button/__test__/button.test.js]: https://github.com/saddlebackdev/react-cm-ui/pull/325/commits/e583e8361bfabd86437159d2720386b9991b0e4f#diff-84e959d96accf1625e2aed89ccb73db78e5e20c9352a392efd06a4edaf09f3ccL91-R91

_MUCH_ better ... thank you!
Resolving.

File: src/inputs/button/buttonConstants.js
Comment (cbrewer1689) [reply src/inputs/button/buttonConstants.js:88]: fixed.

File: docs/src/app/home.jsx
Comment (JLeeC) [inline docs/src/app/home.jsx:137]: Can we a Comment here indicating that this is stop gap only until we can truly deprecate all old buttons, at which point this goes away.  Ideally we would like to replace it all .. but obviously that's gonna cause risk.   Would also like to see a ticket for this somewhere

File: docs/src/global/exampleFrame.jsx
Comment (groberts314) [reply docs/src/global/exampleFrame.jsx:14]: Resolving this to de-clutter the PR.

**Review by JLeeC — APPROVED** [review]
Looks pretty good to me.

File: src/dataDisplay/icon/iconConstants.js
Comment (groberts314) [inline src/dataDisplay/icon/iconConstants.js:7]: **_Not a Blocker_**
Just wondering why separate this out versus define it inline with the others.

File: docs/src/app/home.jsx
Comment (cbrewer1689) [reply docs/src/app/home.jsx:137]: Apologies for this question, but what specifically are we talking about here?

File: src/inputs/button/buttonConstants.js
Comment (groberts314) [inline src/inputs/button/buttonConstants.js:11]: NAB: Do we need a documentation comment on `as` prop?

File: src/inputs/button/buttonConstants.js
Comment (groberts314) [reply src/inputs/button/buttonConstants.js:88]: Resolving.

**Review by groberts314 — APPROVED** [review]
Mostly looks good.  All NABs and questions I believe.

File: docs/src/app/home.jsx
Comment (groberts314) [reply docs/src/app/home.jsx:137]: Maybe add a documentation comment here...?
https://github.com/saddlebackdev/react-cm-ui/pull/325/files#diff-dc9b8e205a6f63db01603b18a6d1f8df41862cd6463bdf63115d839cc9ee2bcbR8

File: docs/src/global/example.jsx
Comment (cbrewer1689) [reply docs/src/global/example.jsx:8]: I believe so, it's belong so I am not sure how to answer. :D

File: src/dataDisplay/icon/iconConstants.js
Comment (cbrewer1689) [reply src/dataDisplay/icon/iconConstants.js:7]: I don't know why. I thought the same thing, but never came back to it.

File: src/inputs/button/buttonConstants.js
Comment (cbrewer1689) [reply src/inputs/button/buttonConstants.js:11]: Sure, i can do this.

**Comment by cbrewer1689** [comment]
"Do Not Merge" because we `dev` is to be 10.1.6. When this version is release we can then merge this PR.

File: docs/src/global/example.jsx
Comment (groberts314) [reply docs/src/global/example.jsx:8]: All good.  As long as ESLINT is appeased, I'm good! =)
Resolving.

File: src/dataDisplay/icon/iconConstants.js
Comment (groberts314) [reply src/dataDisplay/icon/iconConstants.js:7]: **_Not a Blocker_**
We can move it inline, or we can leave it as-is.  Your call.

**Pull request merged**
